toolkit/tools/specreader/specreader.go

```diff
@@ -200,8 +244,44 @@ func createChroot(workerTar, buildDir, specsDir, srpmsDir string) (chroot *safec
 	return
 }
 
+func findSpecFiles(specsDir string, specListSet map[string]bool) (specFiles []string, err error) {
+	// Find the filepath for each spec in the SPECS directory.
+	if len(specListSet) == 0 {
+		var specSearch string
+		specSearch, err = filepath.Abs(filepath.Join(specsDir, "**/*.spec"))
+		if err != nil {
+			err = fmt.Errorf("invalid spec path: '%s'. Error: %w", specsDir, err)
+			return nil, err
+		}
+		specFiles, err = filepath.Glob(specSearch)
+		if err != nil {
+			err = fmt.Errorf("failed to find *.spec files. Check that %s is the correct directory. Error: %w", specsDir, err)
+			return nil, err
+		}
+		return
+	} else {
+		for specName := range specListSet {
+			var matchingSpecFiles []string
+
+			specSearch := filepath.Join(specsDir, fmt.Sprintf("**/%s.spec", specName))
+			matchingSpecFiles, err = filepath.Glob(specSearch)
+
+			// If a SPEC is in the parse list, it should be parsed.
+			if err != nil {
+				return nil, err
+			}
+			if len(matchingSpecFiles) != 1 {
+				err = fmt.Errorf("unexpected number of matches (%d) for spec file (%s)", len(matchingSpecFiles), specName)
```

Comment:
Nit-pick: let's be consistent and use either brackets or single quotes when surrounding file paths.

Reply:
```suggestion
				err = fmt.Errorf("unexpected number of matches '%d' for spec file '%s'", len(matchingSpecFiles), specName)
```